### PR TITLE
bpf: Derive proxy_port from policy rather than CT

### PR DESCRIPTION
**Pull request by joestringer**
Trust the policy's determination of proxy_port rather than using the
conntrack entry for determining whether to send to the proxy in the
forward direction. This should allow traffic to fail over to via proxy
or not via proxy quickly when the policy is changed, rather than waiting
for an out-of-band userspace thread to iterate through the conntrack
entries and update the conntrack entry with the correct proxy_port.

* [x] TODO: Fix up the commit message

Signed-off-by: Joe Stringer <joe@covalent.io>

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: bpf/bpf_lxc.c [hunk 1]
```diff
@@ -236,6 +240,8 @@ static inline int ipv6_l3_from_lxc(struct __sk_buff *skb,
 	case CT_REPLY:
 		policy_mark_skip(skb);
 
+		if (ct_state.proxy_port)
+			verdict = ct_state.proxy_port;
```
Comment (joestringer) [inline bpf/bpf_lxc.c]: This part should be unnecessary. On egress reply, if the connection is already going through the proxy then the connection will be between the host and the endpoint, so we don't need to do special handling to redirect to proxy.

**Comment by joestringer** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by tgraf** [comment]
Closing in favor of #3566

**Pull request closed without merging**